MekHQ/src/mekhq/campaign/mission/BotForce.java

```diff
@@ -125,12 +128,34 @@ public void setName(String name) {
     }
 
     public List<Entity> getEntityList() {
-        return entityList;
+        return Collections.unmodifiableList(entityList);
+    }
+
+    public boolean removeEntity(int index) {
+        Entity e = null;
+        if ((index >= 0) && (index < entityList.size())) {
+            e = entityList.remove(index);
+            nameTracker.remove(e, updated -> {
+                updated.generateShortName();
+                updated.generateDisplayName();
+            });
+        }
+
+        return e != null;
     }
 
     public void setEntityList(List<Entity> entityList) {
-        // Filter all nulls out of the parameter entityList
-        this.entityList = entityList.stream().filter(Objects::nonNull).collect(Collectors.toCollection(ArrayList::new));
+        nameTracker.clear();
+
+        List<Entity> entities = new ArrayList<>();
+        for (Entity e : entities) {
```

Comment:
This doesn't do anything

Reply:
You're correct, too much deleting and moving.